internal/handler/userHandler.go

```diff
@@ -82,24 +94,34 @@ func (h *Handler) Save(context *gin.Context) {
 // @Description Find user by provided id
 // @Tags Users
 // @Produce json
-// @Param id path int true "User id"
-// @Success 201 {object} response.UserResponseDto "User Found successfully"
-// @Failure 422 {object} response.ErrorDto "Id in the wrong format"
+// @Param id path string true "User id"
+// @Failure 422 {object} response.ErrorDto "ID in the wrong format"
 // @Failure 404 {object} response.ErrorDto "User not found"
+// @Failure 500 {object} response.ErrorDto "Failed to connect to database"
 // @Router /users/{id} [get]
 func (h *Handler) FindById(context *gin.Context) {
 	id := context.Param("id")
 	logger.Infof("Finding user by id %s", id)
 	user, err := h.service.GetUserById(context, id)
-	if err != nil {
+	if user == nil {
+		var databaseError *exceptions.DatabaseError
+		if errors.As(err, &databaseError) {
+			logger.Errorf("Error while connecting to database. %v", err)
+			errorResponse := response.ErrorDto{
+				Message:   "Something went wrong",
+				Timestamp: time.Now(),
+			}
+			context.IndentedJSON(http.StatusInternalServerError, errorResponse)
+			return
+		}
 		logger.Errorf("Error while finding user by id %s. %v", id, err)
```

Comment:
We should not log an error here since it will always be nil

Reply:
Fixed